torchx/mix.exs

```diff
@@ -19,7 +19,14 @@ defmodule Torchx.MixProject do
       compilers: compilers() ++ Mix.compilers(),
       elixirc_paths: elixirc_paths(Mix.env()),
       aliases: aliases(),
-      make_env: %{"LIBTORCH_DIR" => System.get_env("LIBTORCH_DIR", @libtorch_cache)}
+      make_env: %{
+        "LIBTORCH_DIR" => System.get_env("LIBTORCH_DIR", @libtorch_cache),
+        "BUILD_EMBEDDED" =>
+          System.get_env(
+            "BUILD_EMBEDDED",
+            "#{Keyword.get(Mix.Project.config(), :build_embedded)}"
```

Comment:
Doesn't this render "nil" if `:build_embedded` is not defined in `Mix.Project.config()`?

Reply:
Hi @polvalente, this default value of this variable is set to `false` by `elixir_make`. At least, it is said so in its inline docs [here](https://github.com/elixir-lang/elixir_make/blob/84655d2643796bf35489fd2b6936ddc5e904b913/lib/mix/tasks/compile.make.ex#L90).

Reply:
Cool. I'd still invert the test to check for "true" because this will make the behavior closer to Elixir, in which "false" and "nil" will test as falsy

Reply:
There is another option here, which is to extract the archive directly to priv, instead of the cache directory. This way we get rid of the symlink altogether. WDYT @polvalente?

Reply:
Oh, I didn't see this comment in time for the merge.
This only solves the issue when we're using autodownload, though. The solution in this PR will also work when we're dealing with user-provided LIBTORCH_DIR values as well (in which we don't use the cache dir)